gdx/src/com/badlogic/gdx/math/Intersector.java

```diff
@@ -82,7 +124,9 @@ public static boolean isPointInTriangle (float px, float py, float ax, float ay,
 	}
 
 	public static boolean intersectSegmentPlane (Vector3 start, Vector3 end, Plane plane, Vector3 intersection) {
-		Vector3 dir = v0.set(end).sub(start);
+		final IntersectorData data = tlData.get();
```

Comment:
What happens if you call this method in parallel threads? How they will use different structures?

Reply:
Thread local storage is very useful. It's basically a piece of memory belonging to a specific thread.
So.. if you call this method in parallel threads, each thread will refer to its own instance of _IntersectorData_